src/coreclr/jit/gentree.h

```diff
@@ -8343,6 +8347,43 @@ inline bool GenTree::IsIntegralConst() const
 #endif // !TARGET_64BIT
 }
 
+//-------------------------------------------------------------------------
+// IsIntegralConstPow2: Determines whether the node is an integral constant
+//                      of the power of 2.
+//
+// Arguments:
+//     None
+//
+// Return Value:
+//     Returns true if this GenTree is an integral constant of the power
+//     of 2.
+inline bool GenTree::IsIntegralConstPow2() const
+{
+    return IsIntegralConst() && isPow2(AsIntCon()->IconValue());
```

Comment:
Otherwise it'll fail for 32 bit targets.

Reply:
Side note: this function is not as precise as it could be, since the constant nodes store values sign-extended. It means all `uint` powers of two `> int.MaxValue` will return `false` from it. That should be documented, or fixed.

Reply:
Ok - to make sure I understand what you mean - the value stored in these constant nodes will always be signed right?
If that's the case, I can just use `IsIntegralConstAbsPow2` instead.

Reply:
Sorry for coming back to you this late -- the mail took its time.

Yes, the issue was with the corner case of 2^31, when viewed as unsigned, it is a power of two, but not when signed.

>  If that's the case, I can just use `IsIntegralConstAbsPow2` instead.

I am not convinced this is correct. Consider `-2`: `abs(-2)` is a power of two, but `-2` as unsigned is `4294967294` (for `TYP_INT`), which is not.

Reply:
I see - that makes sense. I will try to address this.